### PR TITLE
[scripts] cleanup release script and make it more robust

### PR DESCRIPTION
**Pull request by mxm**
- fail on command error
- make Mac OS X compatible
- fix uberjar file limit detection
- cleanup

### CODE REVIEW

File: tools/create_release_files.sh [hunk 1]
```diff
@@ -36,57 +58,59 @@ if [[ `basename $CURR_DIR` != "tools" ]] ; then
...
 sonatype_pw=${sonatype_pw:-XXX}
 
-#echo $NEW_VERSION_HADOOP1
-#sleep 5
-#set -e
 
-# create source package
+if [ "$(uname)" == "Darwin" ]; then
+    SHASUM=shasum
+    MD5SUM=md5
```
Comment (uce) [inline tools/create_release_files.sh]: this has to be `md5 -r`. on my machine, md5 produces
```
$md5 pom.xml 
MD5 (pom.xml) = 7ed5618e87c804233d5e4f18ab7aa7a6
```
whereas the `md5sum` behaviour is
```
Adding the `-r` flag reverses the output, which produces the expected result.
$ md5sum pom.xml
4894008e9ef41fa2925e3596b07ce1d7  pom.xml
```

File: tools/create_release_files.sh [hunk 2]
```diff
@@ -36,57 +58,59 @@ if [[ `basename $CURR_DIR` != "tools" ]] ; then
...
+sonatype_user=${sonatype_user:-yourapacheidhere}
 sonatype_pw=${sonatype_pw:-XXX}
 
-#echo $NEW_VERSION_HADOOP1
-#sleep 5
-#set -e
 
-# create source package
+if [ "$(uname)" == "Darwin" ]; then
+    SHASUM=shasum
```
Comment (uce) [inline tools/create_release_files.sh]: this has to be `shasum -a 512`. default algorithm on my machine is sha1sum.

```
$ shasum pom.xml       
31545bb4b33796762112bb4483fdecc943450e1e  pom.xml
```
vs. the expected
```
$ shasum -a 512 pom.xml 
f96216189b32dee88fb0d00f75337b7a21508a92db6f6df440448c10159b54c7f00e19faeae3de5a53a829b7223253708a168f0a13bbde028e52fab9d0e1d6db  pom.xml
```

**Comment by uce** [comment]
changes look good modulo inline comments

File: tools/create_release_files.sh (hunk 2 above)
Comment (mxm) [inline tools/create_release_files.sh]: Yes, actually I thought the default is SHA256 which is sufficient but it is SHA-1.

https://www.apache.org/dev/release-signing.html#sha-checksum

File: tools/create_release_files.sh (hunk 1 above)
Comment (mxm) [inline tools/create_release_files.sh]: Where does it state, this is the desired output? For consistency, I probably that makes sense.

**Comment by mxm** [comment]
Thanks for the review.

File: tools/create_release_files.sh (hunk 1 above)
Comment (uce) [inline tools/create_release_files.sh]: I didn't know that md5sum correctly works with the reversed files as well. So this is optional.

Comment (mxm) [inline tools/create_release_files.sh]: Well, the important thing is to have the checksum to be able to verify with
any kind of md5 calculation tool. But I changed it for consistency. Thanks!

**Comment by uce** [comment]
Is there anything blocking this? If not, feel free to go ahead and merge this. :)

**Comment by mxm** [comment]
Thanks, merging then.

**Pull request closed without merging**